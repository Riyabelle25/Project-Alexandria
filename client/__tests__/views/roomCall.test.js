import Video from '../../src/views/roomCall';
import React from "react";
import { shallow, mount, configure } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'; // Smart components

configure({ adapter: new Adapter() });

describe('<Video />', () => {
  const initialState = {user: null}
  const mockStore = configureStore()
  const props = {match: {name: "BookClub"}};
  let store
  describe('render()', () => {
    test('renders the component', () => {
      store = mockStore(initialState)
      const wrapper = shallow(<Provider store={store}><Video {...props} /></Provider>);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});

test('check if video and audio work on giving permissions', () => {
   
    const props = {match: {name: "BookClub"}};
    const wrapper = mount(<Video {...props} />)
    expect(wrapper.instance().state.roomName).toBe("BookClub")
    expect(
        (wrapper.instance().state.video && wrapper.instance().videoAvailable) 
            || (wrapper.instance().state.audio && wrapper.instance().audioAvailable)).toBe(false)
    // wrapper.find('button.counter-button').simulate('click')
    // wrapper.setState({count: 1})
    wrapper.instance().connect()
    expect(wrapper.instance().state.askForUsername).toBe(false)
  })