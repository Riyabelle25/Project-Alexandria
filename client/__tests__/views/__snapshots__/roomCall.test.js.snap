// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> render() renders the component 1`] = `
<Video
  match={
    Object {
      "name": "BookClub",
    }
  }
/>
`;
